pkg/controller/job/job_controller.go

```diff
@@ -1453,17 +1509,25 @@ func (jm *Controller) manageJob(ctx context.Context, job *batch.Job, jobCtx *syn
 			diff = int32(MaxPodCreateDeletePerSync)
 		}
 
+		var indexesToAdd []int
+		if isIndexedJob(job) {
+			indexesToAdd = firstPendingIndexes(jobCtx, int(diff), int(*job.Spec.Completions))
+			if hasBackoffLimitPerIndex(job) {
+				indexesToAdd, remainingTime = jm.getPodCreationInfoForIndependentIndexes(logger, indexesToAdd, jobCtx.podsWithDelayedDeletionPerIndex)
+				if remainingTime > 0 {
+					jm.enqueueSyncJobWithDelay(logger, job, remainingTime)
+					return 0, metrics.JobSyncActionPodsCreated, nil
+				}
+			}
+			diff = int32(len(indexesToAdd))
+		}
+
 		jm.expectations.ExpectCreations(logger, jobKey, int(diff))
 		errCh := make(chan error, diff)
 		logger.V(4).Info("Too few pods running", "key", jobKey, "need", wantActive, "creating", diff)
 
 		wait := sync.WaitGroup{}
 
-		var indexesToAdd []int
-		if isIndexedJob(job) {
-			indexesToAdd = firstPendingIndexes(jobCtx, int(diff), int(*job.Spec.Completions))
-			diff = int32(len(indexesToAdd))
```

Comment:
oh, this should always have been before calling `ExpectCreations`, although in general there shouldn't be duplicated indexes.

Reply:
yeah, I thought so, should we have a prep PR before this one? It would also make the diff smaller, and could be cherry-picked back if needed.

Reply:
Opened just in case: https://github.com/kubernetes/kubernetes/pull/119335